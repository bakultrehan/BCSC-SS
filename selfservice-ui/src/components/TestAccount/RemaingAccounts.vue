/** * RemaingAccounts component */

<template>
  <div
    class="remaining-acc-txt"
    v-if="getTestAccountCount && getTestAccountCount.total"
  >
    <span class="count" :class="getTestAccountCount.warning && 'not-available'">
      {{ getTestAccountCount.available }}</span
    ><span class="divide">/</span
    ><span class="total">{{ getTestAccountCount.total }}</span>
    <span class="remaining-text">{{
      $t('dashboard.txtTestAccountRemain')
    }}</span>
  </div>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';

const TestAccountModule = namespace('TestAccountModule');

@Component
export default class RemaingAccounts extends Vue {
  @TestAccountModule.Getter('getTestAccountCount')
  public getTestAccountCount!: any;
  @TestAccountModule.Action('loadTestAccountCount')
  public loadTestAccountCount!: any;

  private mounted() {
    this.loadTestAccountCount();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.remaining-acc-txt {
  color: $BCgovFontColorInvertedDark;
  & .count {
    font-size: 48px;
    font-weight: bold;
    color: $succssGreen;
    &.not-available {
      color: $BCgovInputError;
    }
  }
  & .total {
    font-size: 36px;
    font-weight: bold;
  }
  & .divide {
    font-size: 36px;
    font-weight: bold;
    margin: 0 6px;
  }
  & .remaining-text {
    font-weight: bold;
    font-size: 22px;
    margin-left: 10px;
  }
}
</style>
