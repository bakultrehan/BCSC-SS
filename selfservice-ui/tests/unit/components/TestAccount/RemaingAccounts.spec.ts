import { shallowMount } from '@vue/test-utils';
import RemaingAccounts from '@/components/TestAccount/RemaingAccounts.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('RemaingAccounts.vue', () => {
  const store = new Vuex.Store({
    modules: {
      TestAccountModule: {
        namespaced: true,
        state: {},
        getters: {
          getTestAccountCount: jest.fn(),
          isLoading: jest.fn(),
        },
        actions: {
          loadTestAccountCount: jest.fn(),
          clearStatus: jest.fn(),
        },
      },
    },
  });

  const mountFunction = (options: any) => {
    return shallowMount(RemaingAccounts, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options,
    });
  };

  it('renders props when passed with gettors', () => {
    const RemaingAccountsPage = mountFunction({});
    expect(RemaingAccountsPage.element).toMatchSnapshot();
  });
});
