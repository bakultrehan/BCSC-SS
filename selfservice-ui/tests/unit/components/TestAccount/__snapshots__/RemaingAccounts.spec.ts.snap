// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemaingAccounts.vue renders props when passed with gettors 1`] = `<!---->`;
